File: JANUARY/10-01-2026/712.go
func minimumDeleteSum(s1 string, s2 string) int {
    n, m := len(s1), len(s2)
    dp := make([][]int, n+1)
    for i := range dp {
        dp[i] = make([]int, m+1)
    }

    for i := n - 1; i >= 0; i-- {
        dp[i][m] = dp[i+1][m] + int(s1[i])
    }

    for j := m - 1; j >= 0; j-- {
        dp[n][j] = dp[n][j+1] + int(s2[j])
    }

    for i := n - 1; i >= 0; i-- {
        for j := m - 1; j >= 0; j-- {
            if s1[i] == s2[j] {
                dp[i][j] = dp[i+1][j+1]
            } else {
                a := int(s1[i]) + dp[i+1][j]
                b := int(s2[j]) + dp[i][j+1]
                if a < b {
                    dp[i][j] = a
                } else {
                    dp[i][j] = b
                }
            }
        }
    }

    return dp[0][0]
}

File: JANUARY/10-01-2026/712.js
var minimumDeleteSum = function(s1, s2) {
    const n = s1.length, m = s2.length;
    const dp = Array.from({ length: n + 1 }, () => Array(m + 1).fill(0));

    for (let i = n - 1; i >= 0; i--)
        dp[i][m] = dp[i + 1][m] + s1.charCodeAt(i);

    for (let j = m - 1; j >= 0; j--)
        dp[n][j] = dp[n][j + 1] + s2.charCodeAt(j);

    for (let i = n - 1; i >= 0; i--) {
        for (let j = m - 1; j >= 0; j--) {
            if (s1[i] === s2[j])
                dp[i][j] = dp[i + 1][j + 1];
            else
                dp[i][j] = Math.min(
                    s1.charCodeAt(i) + dp[i + 1][j],
                    s2.charCodeAt(j) + dp[i][j + 1]
                );
        }
    }

    return dp[0][0];
};